pycromanager/acq_util.py

```diff
@@ -181,7 +181,14 @@ def multi_d_acquisition_events(
         z_positions = xyz_positions[:, 2][:, None]
 
     if has_zsteps:
-        z_rel = np.arange(z_start, z_end + z_step, z_step)
+        z_step = abs(z_step) 
+        if z_start < z_end: 
+            z_rel = np.arange(z_start, z_end + z_step, z_step) 
+            
+        else:   
+            z_rel = np.arange(z_start, z_end , -z_step)
```

Comment:
We should expect to get the same number of events whether `z_start < z_end` or not. I think you'll need to modify the else statement to `z_rel = np.arange(z_start, z_end - z_step, -z_step)` and change `test_positive_z_step_direction` accordingly.